proto/penumbra/penumbra/core/component/dex/v1/dex.proto

```diff
@@ -68,8 +68,10 @@ message SwapBody {
   // in the future we'll want separate commitments to each delta_j_i
   // anyways in order to prove consistency with flow encryption.
   asset.v1.BalanceCommitment fee_commitment = 4;
+  // A commitment to the value balance.
+  asset.v1.BalanceCommitment balance_commitment = 5;
```

Comment:
We can't change protobuf like that unfortunately, this is a very breaking change.

Here are a couple useful links about protobuf versioning:
- the most important one: [protos dos and donts](https://protobuf.dev/programming-guides/dos-donts/)
- https://github.com/uber/prototool/blob/dev/style/README.md#package-versioning
- https://buf.build/docs/breaking/rules

Reply:
The right way to do this is to add a new field, without changing the numbering so that it doesn't affect the binary encoding of the messages.